### PR TITLE
Replace FileSystemContext in presto-raptor by HdfsContext

### PR DESCRIPTION
**Pull request by fvcortes**
This PR is intended to fix [ #14011](https://github.com/prestodb/presto/issues/14011) by replacing the use of FileSystemContext in presto-raptor by HdfsContext since presto-hive-metastore has been separated into a standalone module.

== NO RELEASE NOTE ==

### CODE REVIEW

**Comment by highker** [comment]
Can we squash all commits into one?

**Comment by fvcortes** [comment]
Travis build jobs passed but one in test-hive-repartitioning. I don't know if it has something to do with my changes:
image

File: presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorPageSinkProvider.java [hunk 1]
```diff
@@ -59,7 +59,7 @@ public ConnectorPageSink createPageSink(ConnectorTransactionHandle transactionHa
 
         RaptorOutputTableHandle handle = (RaptorOutputTableHandle) tableHandle;
         return new RaptorPageSink(
-                new FileSystemContext(session),
+                new HdfsContext(session.getIdentity()),
```
Comment (highker) [inline presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorPageSinkProvider.java:62]: We also need all other information other than identity. Schema and table are available in `handle`. Same for all other places.

File: presto-raptor/src/test/java/com/facebook/presto/raptor/storage/TestOrcStorageManager.java [hunk 2]
```diff
@@ -140,7 +141,7 @@
     private static final DataSize MAX_FILE_SIZE = new DataSize(1, MEGABYTE);
     private static final Duration MISSING_SHARD_DISCOVERY = new Duration(5, TimeUnit.MINUTES);
     private static final ReaderAttributes READER_ATTRIBUTES = new ReaderAttributes(new DataSize(1, MEGABYTE), new DataSize(1, MEGABYTE), new DataSize(1, MEGABYTE), new DataSize(1, MEGABYTE), true, false);
-
```
Comment (highker) [inline presto-raptor/src/test/java/com/facebook/presto/raptor/storage/TestOrcStorageManager.java:143]: leave an empty line between constants and variables

File: presto-raptor/src/test/java/com/facebook/presto/raptor/storage/organization/TestShardCompactor.java [hunk 3]
```diff
@@ -80,7 +81,7 @@
     private static final int MAX_SHARD_ROWS = 1000;
     private static final PagesIndexPageSorter PAGE_SORTER = new PagesIndexPageSorter(new PagesIndex.TestingFactory(false));
     private static final ReaderAttributes READER_ATTRIBUTES = new ReaderAttributes(new DataSize(1, MEGABYTE), new DataSize(1, MEGABYTE), new DataSize(1, MEGABYTE), new DataSize(1, MEGABYTE), true, false);
-
```
Comment (highker) [inline presto-raptor/src/test/java/com/facebook/presto/raptor/storage/organization/TestShardCompactor.java:83]: same

File: presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorPageSinkProvider.java (hunk 1 above)
Comment (fvcortes) [reply presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorPageSinkProvider.java:62]: Got it. I see that I can get schema and table names from RaptorOutputTableHandle:
```suggestion
                new HdfsContext(session, handle.getSchemaName(), handle.getTableName())
```

File: presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorPageSinkProvider.java [hunk 4]
```diff
@@ -82,7 +82,7 @@ public ConnectorPageSink createPageSink(ConnectorTransactionHandle transactionHa
 
         RaptorInsertTableHandle handle = (RaptorInsertTableHandle) tableHandle;
         return new RaptorPageSink(
-                new FileSystemContext(session),
+                new HdfsContext(session.getIdentity()),
```
Comment (fvcortes) [inline presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorPageSinkProvider.java:85]: But schema and table names are not available in RaptorInsertTableHandle.

File: presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorPageSourceProvider.java [hunk 5]
```diff
@@ -69,7 +69,7 @@ public ConnectorPageSource createPageSource(
         Optional<Map<String, Type>> columnTypes = raptorSplit.getColumnTypes();
         boolean tableSupportsDeltaDelete = raptorSplit.isTableSupportsDeltaDelete();
 
-        FileSystemContext context = new FileSystemContext(session);
+        HdfsContext context = new HdfsContext(session.getIdentity());
```
Comment (fvcortes) [inline presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorPageSourceProvider.java:72]: So does here.

Comment (highker) [reply presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorPageSourceProvider.java:72]: We can ignore schema and table for page source/sink provider at the moment

File: presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorPageSinkProvider.java (hunk 4 above)
Comment (highker) [reply presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorPageSinkProvider.java:85]: We can ignore schema/table. But we should populate other fields.

File: presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorPageSourceProvider.java (hunk 5 above)
Comment (highker) [reply presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorPageSourceProvider.java:72]: Same

File: presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorPageSinkProvider.java (hunk 4 above)
Comment (highker) [inline presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorPageSinkProvider.java:85]: Could you address the comments https://github.com/prestodb/presto/pull/14680/files#r443993637? We need more than `identity`.

Comment (fvcortes) [reply presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorPageSinkProvider.java:85]: Should I populate the fields with an empty string?

Comment (highker) [reply presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorPageSinkProvider.java:85]: Create a new interface `public HdfsContext(ConnectorSession session)` for `HdfsContext`

Comment (highker) [reply presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorPageSinkProvider.java:85]: BTW, the constructors for HdfsContext is really bad. We should have something like this
```java
    public HdfsContext(ConnectorSession session)
    {
        this(session, Optional.empty(), Optional.empty());
    }

    public HdfsContext(ConnectorSession session, String schemaName)
    {
        this(session, Optional.of(schemaName), Optional.empty());
    }

    public HdfsContext(ConnectorSession session, String schemaName, String tableName)
    {
        this(session, Optional.of(schemaName), Optional.of(tableName));
    }

    private HdfsContext(ConnectorSession session, Optional<String> schemaName, Optional<String> tableName)
    {
        requireNonNull(session, "session is null");
        this.identity = requireNonNull(session.getIdentity(), "session.getIdentity() is null");
        this.source = requireNonNull(session.getSource(), "session.getSource() is null");
        this.queryId = Optional.of(session.getQueryId());
        this.schemaName = requireNonNull(schemaName, "schemaName is null");
        this.tableName = requireNonNull(tableName, "tableName is null");
        this.clientInfo = session.getClientInfo();
    }

Comment (fvcortes) [reply presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorPageSinkProvider.java]: Ok! Doing an explicit constructor invocation makes it clearer.

File: presto-hive-common/src/main/java/com/facebook/presto/hive/HdfsContext.java [hunk 6]
```diff
@@ -40,28 +40,29 @@ public HdfsContext(ConnectorIdentity identity)
...
-        this.queryId = Optional.of(session.getQueryId());
-        this.schemaName = Optional.of(schemaName);
-        this.tableName = Optional.empty();
-        this.clientInfo = session.getClientInfo();
+        this(session, Optional.of(schemaName), Optional.empty());
     }
 
     public HdfsContext(ConnectorSession session, String schemaName, String tableName)
+    {
+        this(session, Optional.of(schemaName), Optional.of(tableName));
```
Comment (highker) [inline presto-hive-common/src/main/java/com/facebook/presto/hive/HdfsContext.java:55]: requireNonNull for schemaName and tableName

File: presto-hive-common/src/main/java/com/facebook/presto/hive/HdfsContext.java [hunk 7]
```diff
@@ -40,28 +40,29 @@ public HdfsContext(ConnectorIdentity identity)
...
     {
-        requireNonNull(session, "session is null");
-        requireNonNull(schemaName, "schemaName is null");
-        this.identity = requireNonNull(session.getIdentity(), "session.getIdentity() is null");
-        this.source = requireNonNull(session.getSource(), "session.getSource()");
-        this.queryId = Optional.of(session.getQueryId());
-        this.schemaName = Optional.of(schemaName);
-        this.tableName = Optional.empty();
-        this.clientInfo = session.getClientInfo();
+        this(session, Optional.of(schemaName), Optional.empty());
```
Comment (highker) [inline presto-hive-common/src/main/java/com/facebook/presto/hive/HdfsContext.java:50]: requireNonNull for schemaName

File: presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorPageSinkProvider.java [hunk 8]
```diff
@@ -59,7 +59,7 @@ public ConnectorPageSink createPageSink(ConnectorTransactionHandle transactionHa
 
         RaptorOutputTableHandle handle = (RaptorOutputTableHandle) tableHandle;
         return new RaptorPageSink(
-                new FileSystemContext(session),
+                new HdfsContext(session),
```
Comment (highker) [inline presto-raptor/src/main/java/com/facebook/presto/raptor/RaptorPageSinkProvider.java:62]: Pass in schema and table as well from `handle`

File: presto-raptor/src/main/java/com/facebook/presto/raptor/filesystem/HdfsStorageService.java [hunk 9]
```diff
@@ -118,7 +121,7 @@ private void createDirectory(Path directory)
     {
         boolean madeDirectory;
         try {
-            FileSystem fileSystem = environment.getFileSystem(FileSystemContext.DEFAULT_RAPTOR_CONTEXT);
```
Comment (highker) [inline presto-raptor/src/main/java/com/facebook/presto/raptor/filesystem/HdfsStorageService.java:121]: move DEFAULT_RAPTOR_CONTEXT to FileSystemUtil

File: presto-hive-common/src/main/java/com/facebook/presto/hive/HdfsContext.java (hunk 6 above)
Comment (fvcortes) [reply presto-hive-common/src/main/java/com/facebook/presto/hive/HdfsContext.java:55]: Wouldn't requireNonNull already in the last constructor be enough?

`this.schemaName = requireNonNull(schemaName, "schemaName is null");`
`this.tableName = requireNonNull(tableName, "tableName is null");`

Comment (highker) [reply presto-hive-common/src/main/java/com/facebook/presto/hive/HdfsContext.java:55]: They are completely different. This is of type `String`; the other is of type `Optional<String>`. A null input here will throw NPE.

Comment (fvcortes) [reply presto-hive-common/src/main/java/com/facebook/presto/hive/HdfsContext.java:55]: Understood, thanks.

File: presto-raptor/src/main/java/com/facebook/presto/raptor/filesystem/HdfsStorageService.java (hunk 9 above)
Comment (fvcortes) [reply presto-raptor/src/main/java/com/facebook/presto/raptor/filesystem/HdfsStorageService.java:121]: Should I move it as a public or private with getter?

**Pull request merged**